### DISCUSSION TITLE
Test file with parametrized dynamic import doesn't work

### DISCUSSION THREAD

**Discussion by piothPaproKee** [cypress-io/cypress#28620] [Questions and Help]
Hi everyone

I have an issue when trying to dynamically import a module in my test file. 

```
// test.cy.ts
const file = 'fileName';
import(`../../directory/${file}`)
  .then(...)
  .catch(...);
```

It was working fine with cypress 8, i just upgraded the project to cypress 13 and it doesn't work anymore :<
If i inline the badboy then it works like a charm again:
```
import(`../../directory/fileName`)
  .then(...)
  .catch(...);
  ```
The project uses cucumber and some custom configuration, but i also tried it on simpler, smaller project, with cypress@12 and without cucumber and it doesn't work either.

I have tried webpack eager flag (to avoid code splitting) and several other ideas but nothing worked. 

Any clues? Should it just work and there's something messed up with my config? 

Thanks

**Comment by portyu9** [comment]
Hi @piothPaproKee 

This is a bundler-resolution issue rather than Cypress removing support for `import()`.

Cypress 13 changed the default E2E preprocessor from **Webpack 4 to Webpack 5**. Webpack does support dynamic imports, but it must be able to determine the possible files at build time.

So this:

```js
const file = 'fileName'

import(`../../directory/${file}`)
```

is less deterministic than a normal static import, and differences in Webpack's generated context can explain why the hard-coded path works.

The most robust solution is to make the set of possible modules explicit:

```js
const modules = {
  fileName: () => import('../../directory/fileName.js'),
  anotherFile: () => import('../../directory/anotherFile.js'),
}

const loadModule = modules[file]

if (!loadModule) {
  throw new Error(`Unknown module: ${file}`)
}

loadModule().then((module) => {
  // use module
})
```

Use the actual extension/path from your project.

For a large set of modules you can instead use a Webpack context, but the important part is the same: Webpack must know the directory/file pattern while it is bundling the spec.

Changing the Webpack mode to `eager` does not solve an unresolved module context — it only changes **when** already-resolved modules are loaded.

So I would not downgrade Cypress just for this. Make the dynamic import statically bounded for Webpack 5 instead.

Please mark this answer as accepted if it helped, thank you.
